### DISCUSSION TITLE
Create a command panel as Edge

### DISCUSSION THREAD

**Discussion by uazo** [bromite/bromite#2322] [Ideas]
I would like the browser to give me more information about the current page displayed but currently chromium does not have the necessary ui to show them easily.

The idea is to introduce a scrollable panel as Edge, this:
![image](https://user-images.githubusercontent.com/29201891/189638366-5a5fed51-0dbf-4364-b731-ecb49018c70d.png)

I checked and it is already possible:
![1](https://user-images.githubusercontent.com/29201891/189638963-52ed5e45-69c1-448e-a616-85bbef9b77e1.gif)

The idea is that the panel shows an html+js page with superpowers, able to manage the page and some of its properties and communicate with the browser process. The idea is to use a webui and allow javascript to receive json from the browser and send commands (note that the webui does not have internet access).

for example:
![image](https://user-images.githubusercontent.com/29201891/189643271-48a5192c-a371-4f31-bc10-df3dfb50a891.png)

I would like to evaluate the implementation with you because two ways come to mind:

- ui integrated in bromite, fixed and not modifiable by the user
- ui managed with userscripts, therefore possibly modifiable by the user

The second choice would allow me not to develop the html and js page (which would take me a long time) ...

What do you think?

**Comment by csagan5** [comment]
> The idea is to use a webui and allow javascript to receive json from the browser and send commands (note that the webui does not have internet access).

If the two are not sandboxed separately...then there is risk of security breaches.

> The second choice would allow me not to develop the html and js page (which would take me a long time) ...

If user scripts can both access the webpage and this extra page then the sandbox is already violated.

> The idea is that the panel shows an html+js page with superpowers, able to manage the page and some of its properties and communicate with the browser process.

This part is very vague to me, I have not used Edge in ages so I do not know what you are referring to specifically.

**Reply by uazo** [reply]
certainly are things to check, for now I was interested to know if you would take it into consideration. as soon as I have something working I'll show you.
maybe, I do a issue like "intent to develop" where I explain better.

**Comment by emvaized** [comment]
It would be great to use such UX solution for quickly searching for something by manually inputting the query. Something like "Preview page" feature, but opened manually with gesture like on provided gif. Maybe it would be possible to implement via user scripts as well, with the second approach?
